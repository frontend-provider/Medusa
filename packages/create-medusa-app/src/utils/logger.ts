import winston from "winston"

const consoleTransport = new winston.transports.Console({
  format: winston.format.printf((log) => log.message),
})
const options = {
  transports: [consoleTransport],
}

export const logger = winston.createLogger(options)
