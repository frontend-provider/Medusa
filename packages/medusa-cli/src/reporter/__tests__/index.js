import logger, { Reporter } from "../"

describe(`Reporter`, () => {
  const winstonMock = {
    log: jest.fn(),
  }

  const reporter = new Reporter({
    logger: winstonMock,
    activityLogger: {},
  })

  const getErrorMessages = fn =>
    fn.mock.calls
      .map(([firstArg]) => firstArg)
      .filter(structuredMessage => structuredMessage.level === `error`)

  beforeEach(() => {
    winstonMock.log.mockClear()
  })

  it(`handles "String" signature correctly`, () => {
    reporter.error("Test log")

    const generated = getErrorMessages(winstonMock.log)[0]

    expect(generated).toMatchSnapshot()
  })

  it(`handles "String, Error" signature correctly`, () => {
    reporter.error("Test log", new Error("String Error"))

    const generated = getErrorMessages(winstonMock.log)[0]

    expect(generated).toMatchSnapshot({
      stack: expect.any(Array),
    })
  })

  it(`handles "Error" signature correctly`, () => {
    reporter.error(new Error("Error"))

    const generated = getErrorMessages(winstonMock.log)[0]

    expect(generated).toMatchSnapshot({
      stack: expect.any(Array),
    })
  })
})
