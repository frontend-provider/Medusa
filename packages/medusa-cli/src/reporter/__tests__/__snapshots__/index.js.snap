// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reporter handles "Error" signature correctly 1`] = `
Object {
  "level": "error",
  "message": "Error",
  "stack": Any<Array>,
}
`;

exports[`Reporter handles "String" signature correctly 1`] = `
Object {
  "level": "error",
  "message": "Test log",
}
`;

exports[`Reporter handles "String, Error" signature correctly 1`] = `
Object {
  "level": "error",
  "message": "Test log",
  "stack": Any<Array>,
}
`;
